RecoMET/METFilters/plugins/EcalBadCalibFilter.cc

```diff
@@ -0,0 +1,228 @@
+  , eeRHSrcToken_     (consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("EERecHitSource")))
+  , EBmin_          (iConfig.getParameter<double>("EBminet"))
+  , EEmin_          (iConfig.getParameter<double>("EEminet"))
+  , baddetEB_       (iConfig.getParameter<std::vector<unsigned int> >("baddetEB"))
+  , baddetEE_       (iConfig.getParameter<std::vector<unsigned int> >("baddetEE"))
+  , taggingMode_    (iConfig.getParameter<bool>("taggingMode"))
+  , debug_          (iConfig.getParameter<bool>("debug"))
+{
+  produces<bool>();
+}
+ 
+
+bool EcalBadCalibFilter::filter(edm::StreamID, edm::Event & iEvent, const edm::EventSetup & iSetup) const {
+
+
+  // load required collections
+ 
+ 
+  // EB rechit collection
+  edm::Handle<EcalRecHitCollection> ebRHs;
+  iEvent.getByToken(eeRHSrcToken_, ebRHs);
+ 
+  // EE rechit collection
+  edm::Handle<EcalRecHitCollection> eeRHs;
+  iEvent.getByToken(eeRHSrcToken_, eeRHs);
+ 
+  // Calo Geometry - needed for computing E_t
+  edm::ESHandle<CaloGeometry> pG;
+  iSetup.get<CaloGeometryRecord>().get(pG);
+ 
+  
+  // by default the event is OK
+  bool pass = true;
+ 
+ 
+ 
+  // define energy variables and ix,iy,iz coordinates
+ 
+ 
+  int ix,iy,iz;
+  ix=0,iy=0,iz=0;
+  float ene=0;
+  float et=0;
+ 
+ 
+ 
+  //EE:  loop over the list of bad DetIds (defined in the python file)
+ 
+  //  std::cout << "Starting EE loop" << std::endl;
+
+
+  for (std::vector<unsigned int>::const_iterator eeit = baddetEE_.begin(); eeit != baddetEE_.end(); ++eeit) {
+ 
+    // std::cout << "In EE loop" << std::endl;
+     
+    EEDetId eedet(*eeit);
+ 
+    // std::cout << "got detid" << std::endl;
+ 
+    if (eedet.rawId()==0) continue;
+ 
+    // std::cout << "Searching for DetId=" << eedet.rawId() << std::endl;
+ 
+ 
+    // find rechit corresponding to this DetId
+ 
+    EcalRecHitCollection::const_iterator eehit=eeRHs->find(eedet);
+ 
+    // std::cout << "executed eeRHs->find()" << std::endl;
+ 
+    if (eehit==eeRHs->end()) continue;
+ 
+    // std::cout << "detid found=" << eehit->id().rawId() << std::endl;
+ 
+    // if rechit not found, move to next DetId   
+ 
+    if (eehit->id().rawId()==0 || eehit->id().rawId()!= eedet.rawId()) {
+ 
+      //      std::cout << "Detid " << eedet.rawId() << " not found" << std::endl;
+      continue;
+    
+    }
+     
+ 
+    //  std::cout << "**FOUND Detid " << eedet.rawId() << std::endl;
+ 
+ 
+    // rechit has been found: obtain crystal coordinates, energy 
+ 
+ 
+    ix=eedet.ix();
+    iy=eedet.iy();
+    iz=eedet.zside();
+    ene=eehit->energy();
+ 
+    // compute transverse energy
+     
+    //    std::cout << "accessing geometry" << std::endl;
+ 
+    GlobalPoint posee=pG->getPosition(eedet);
+    float pf=1.0/cosh(posee.eta());
+    et=ene*pf;
+     
+    /*
+      std::cout << "Detid=" << eedet.rawId() 
+      << " ix=" << ix << " iy=" << iy<< " iz=" << iz 
+      << " energy=" << ene << " et=" << et << std::endl;
+    */
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

this applies to all of this PR